### PR TITLE
Add `libshaderc-dev` key

### PR DESCRIPTION
**Pull request by azeey**
Please add the following dependency to the rosdep database.

## Package name:

`libshaderc-dev`

## Package Upstream Source:

https://github.com/google/shaderc/

## Purpose of using this:

Needed by [`gz_ogre_next_vendor`](https://github.com/gazebo-release/gz_ogre_next_vendor)

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/bookworm/libshaderc-dev
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/noble/libshaderc-dev
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/shaderc/libshaderc-devel/
- Arch: https://www.archlinux.org/packages/
  - https://archlinux.org/packages/extra/x86_64/shaderc/
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/media-libs/shaderc
- macOS: https://formulae.brew.sh/
  - skipped
- Alpine: https://pkgs.alpinelinux.org/packages
  - skipped
- NixOS/nixpkgs: https://search.nixos.org/packages
  - skipped
- openSUSE: https://software.opensuse.org/package/
  - skipped
- rhel: https://rhel.pkgs.org/
  - skipped

### CODE REVIEW

**Review by nuclearsandwich — APPROVED** [review]

File: rosdep/base.yaml [hunk 1]
```diff
@@ -5722,6 +5722,14 @@ libsensors4-dev:
...
   debian: [libserial-dev]
   ubuntu: [libserial-dev]
+libshaderc-dev:
+  arch: [shaderc]
+  debian: [libshaderc-dev]
+  fedora: [libshaderc-devel]
+  gentoo: [media-libs/shaderc]
+  ubuntu:
+    '*': null
+    noble: [libshaderc-dev]
```
Comment (clalancette) [inline rosdep/base.yaml:5732]:
```suggestion
  ubuntu:
    '*': [libshaderc-dev]
    focal: null
    jammy: null
```

(so we don't have to update it for future ubuntu distributions)

**Review by clalancette — CHANGES_REQUESTED** [review]
Besides my inline comment, the bot thinks we can satisfy this key on alpine, rhel, and opensuse.

**Comment by azeey** [comment]
> Besides my inline comment, the bot thinks we can satisfy this key on alpine, rhel, and opensuse.

Done in https://github.com/ros/rosdistro/pull/40283/commits/6a3d24c38ab6123436dec6c0c1de0ad7c409f252

**Review by mjcarroll — APPROVED** [review]

File: rosdep/base.yaml [hunk 2]
```diff
@@ -5722,6 +5722,20 @@ libsensors4-dev:
 libserial-dev:
   debian: [libserial-dev]
   ubuntu: [libserial-dev]
+libshaderc-dev:
+  alpine: [shaderc-dev]
+  arch: [shaderc]
+  debian: [libshaderc-dev]
```
Comment (clalancette) [inline rosdep/base.yaml:5728]: Apologies, I missed this on the last review:

```suggestion
  debian:
    '*': [libshaderc-dev]
    bullseye: null
```

**Review by clalancette — CHANGES_REQUESTED** [review]

File: rosdep/base.yaml [hunk 3]
```diff
@@ -5722,6 +5722,22 @@ libsensors4-dev:
...
+  fedora: [libshaderc-devel]
+  gentoo: [media-libs/shaderc]
+  opensuse: [shaderc-devel]
+  rhel:
+    '*': [libshaderc-devel]
+    '8': null
+  ubuntu:
+    '*': [libshaderc-dev]
+    'focal': null
+    'jammy': null
```
Comment (clalancette) [inline rosdep/base.yaml:5740]:
```suggestion
    focal: null
    jammy: null
```

(I'm nitpicking now, but we don't usually use the quotes here)

**Review by clalancette — CHANGES_REQUESTED** [review]

File: rosdep/base.yaml (hunk 3 above)
Comment (azeey) [reply rosdep/base.yaml]: Removed in https://github.com/ros/rosdistro/pull/40283/commits/2487939499fb175dac9d134f1b285b7dc9560f71. I saw the `'8'` for rhel and didn't think to look at other cases for ubuntu.

Comment (clalancette) [reply rosdep/base.yaml]: > Removed in [2487939](https://github.com/ros/rosdistro/commit/2487939499fb175dac9d134f1b285b7dc9560f71). I saw the `'8'` for rhel and didn't think to look at other cases for ubuntu.

Yeah.  It's weird in that you *have* to quote `8` and `*`, but it is optional for strings that start with regular ASCII characters.

**Review by clalancette — APPROVED** [review]

**Pull request merged**